include/repeater_cadmium.hpp

```diff
@@ -0,0 +1,199 @@
+        if (get_messages<typename defs::packet_in>(mbs).size() > 1) {
+            assert(false && "One message at a time");
+        }
+        for (const auto &x : get_messages<typename defs::packet_in>(mbs)) {
+            state.packet = static_cast<int>(x.value);
+            state.sending = true;
+        }
+        if (get_messages<typename defs::ack_in>(mbs).size() > 1) {
+            assert(false && "One message at a time");
+        }
+        for (const auto &x : get_messages<typename defs::ack_in>(mbs)) {
+            state.ack_packet = static_cast<int>(x.value);
+            state.ack = true;
+        }
+    }
+
+    /**
+     * it is the function that make call to internal_transition
+     * followed by call to external transition.
+     * @param e time variable
+     * @param mbs message bags
+    */
+    void confluence_transition(TIME e,
+                               typename make_message_bags<input_ports>::type mbs) {
+        internal_transition();
+        external_transition(TIME(), std::move(mbs));
+    }
+
+    /**
+     * Funtion that sends the packet to the respective
+     * output ports depending on the state type.
+     * When in sending state the value of the input is assigned
+     * to the packet and then sent to the mesage bags.
+     * When in acknowledgement state ack packet value is
+     * assignet ot the output and output is pushed to message bags.
+     * @return Message bags
+    */
+    typename make_message_bags<output_ports>::type output() const {
+        typename make_message_bags<output_ports>::type bags;
+        Message_t out;
+        if (state.sending) {
+            out.value = state.packet;
+            get_messages<typename defs::packet_sent_out>(bags).push_back(out);
+        }
+        if (state.ack)
+        {
+            out.value = state.ack_packet;
+            get_messages<typename defs::ack_received_out>(bags).push_back(out);
+        }
+        return bags;
+    }
+
+    /**
+     * Function that sets the next internal transaction time
+     * @return Next internal time
+    */
+    TIME time_advance() const {
+
+        TIME next_internal;
+        if(state.sending || state.ack)  {
+            next_internal=PREPARATION_TIME;
+        }
+        else{
+            next_internal = std::numeric_limits<TIME>::infinity();
+        }
+        return next_internal;
+    }
+
+    /**
+     * Function that outputs packet num and ack num to ostring stream
+     * @param os the ostring stream
+     * @param i  the structure state type
+     * @return os the ostring stream
+    */
+    friend std::ostringstream& operator<<(std::ostringstream& os,
+                                          const typename Repeater<TIME>::state_type& i) {
+        os << "packetNum: " << i.packet <<
+           " & ackpacket: " << i.ack_packet;
+        return os;
+    }
+};
+
+#endif //ALTERNATEBITPROTOCOLSIMULATOR_REPEATER_CADMIUM_H
```

Comment:
The header guard is wrong. It should be same as the file name. i..e, _/_/REPEATER_CADMIUM_HPP_/_/

Reply:
Check the new file it has been updated

Reply:
I see the same.  my apologies if I'm missing something. The comment would be /** _REPEATER_CADMIUM_HPP_ */

Reply:
It has been fixed